ts/Extensions/Annotations/Controllables/ControllableEllipse.ts

```diff
@@ -0,0 +1,297 @@
+import AxisType from '../../../Core/Axis/AxisType';
+import BBoxObject from '../../../Core/Renderer/BBoxObject';
+import ControllableMixin from '../Mixins/ControllableMixin.js';
+import ControllablePath from './ControllablePath.js';
+import MockPointOptions from '../MockPointOptions';
+
+import U from '../../../Core/Utilities.js';
+
+const {
+    merge,
+    defined
+} = U;
+
+/* eslint-disable no-invalid-this, valid-jsdoc */
+
+/**
+ * A controllable ellipse class.
+ *
+ * @requires modules/annotations
+ *
+ * @private
+ * @class
+ * @name Highcharts.AnnotationControllableEllipse
+ *
+ * @param {Highcharts.Annotation} annotation an annotation instance
+ * @param {Highcharts.AnnotationsShapeOptions} options a shape's options
+ * @param {number} index of the Ellipse
+ */
+
+interface EllipseShapeOptions extends Highcharts.AnnotationsShapeOptions {
+    yAxis: number;
+    xAxis: number;
+    ry: number;
+}
+interface EllipseShapeSVGOptions {
+    cx: number;
+    cy: number;
+    rx: number;
+    ry: number;
+    angle: number;
+}
+
+interface ReferencePointsOptions {
+    x: number;
+    y: number;
+}
+
+class ControllableEllipse implements ControllableMixin.Type {
+
+    /* *
+     *
+     *  Static Properties
+     *
+     * */
+
+    /**
+     * A map object which allows to map options attributes to element
+     * attributes.
+     *
+     * @name Highcharts.AnnotationControllableEllipse.attrsMap
+     * @type {Highcharts.Dictionary<string>}
+     */
+    public static attrsMap = merge(ControllablePath.attrsMap, {
+        ry: 'ry'
+    });
+
+    /* *
+     *
+     *  Constructors
+     *
+     * */
+
+    public constructor(
+        annotation: Annotation,
+        options: EllipseShapeOptions,
+        index: number
+    ) {
+        this.init(annotation, options, index);
+        this.collection = 'shapes';
+    }
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public addControlPoints = ControllableMixin.addControlPoints;
+    public anchor = ControllableMixin.anchor;
+    public attr = ControllableMixin.attr;
+    public attrsFromOptions = ControllableMixin.attrsFromOptions;
+    public destroy = ControllableMixin.destroy;
+    public getPointsOptions = ControllableMixin.getPointsOptions;
+    public linkPoints = ControllableMixin.linkPoints;
+    public point = ControllableMixin.point;
+    public scale = ControllableMixin.scale;
+    public setControlPointsVisibility = ControllableMixin.setControlPointsVisibility;
+    public shouldBeDrawn = ControllableMixin.shouldBeDrawn;
+    public transform = ControllableMixin.transform;
+    public translatePoint = ControllableMixin.translatePoint;
+    public transformPoint = ControllableMixin.transformPoint;
+
+    public update(
+        this: ControllableEllipse,
+        newOptions: Highcharts.AnnotationControllableOptionsObject
+    ): void {
+        const options = merge(true, this.options, newOptions);
+        this.options = options;
+        this.redraw();
+    }
```

Comment:
This function seems to be never called (tested: adding from GUI, control points drag&drop, annotation drag&drop, and Popup's edit).